src/NATS.Client.Core/Internal/NatsPipeliningWriteProtocolProcessor.cs

```diff
@@ -230,6 +242,11 @@ private async Task WriteLoopAsync()
         {
             // ignore, will be handled in read loop
         }
+        catch (Exception ex)
```

Comment:
is it worth also handling some socket exceptions as well? e.g. I sometimes see `System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host.` when reconnecting. maybe a warning, since we kind of expecting it on reconnects.

Reply:
Meaning ignore `SocketException` similarly to how we ignore `SocketClosedException`?  I think the socket exceptions bubble up through the read loop, not the write loop

Reply:
we do get it in writer loop too sometimes:
```
2024-01-04 17:26:48.644 +00:00 fail: NATS.Client.Core.Internal.NatsPipeliningWriteProtocolProcessor[0] Unexpected error occured in write loop System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host.    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.CreateException(SocketError error, Boolean forAsyncThrow)    at NATS.Client.Core.Internal.NatsPipeliningWriteProtocolProcessor.WriteLoopAsync() in C:\Users\mtmk\src\nats.net.v2\src\NATS.Client.Core\Internal\NatsPipeliningWriteProtocolProcessor.cs:line 174    at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)    at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)    at System.Threading.ThreadPoolWorkQueue.Dispatch()    at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart() --- End of stack trace from previous location ---    at NATS.Client.Core.Internal.NatsPipeliningWriteProtocolProcessor.WriteLoopAsync() in C:\Users\mtmk\src\nats.net.v2\src\NATS.Client.Core\Internal\NatsPipeliningWriteProtocolProcessor.cs:line 174
```

it's quite rare. we can leave it as it is for this PR. looking at main I think this was the case already.